pkg/diagnose/deployment_metrics.go

```diff
@@ -0,0 +1,92 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnose
+
+import (
+	"strings"
+
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/subctl/pkg/cluster"
+)
+
+const (
+	gwCurlMetricsCommand = "curl -I -m 10 submariner-gateway-metrics.submariner-operator.svc.cluster.local:8080/metrics"
+	gnCurlMetricsCommand = "curl -I -m 10 submariner-globalnet-metrics.submariner-operator.svc.cluster.local:8081/metrics"
+)
+
+func CheckMetricsConfig(clusterInfo *cluster.Info, status reporter.Interface) bool {
+	status.Start("Checking if the metrics are accessible from non-gateway nodes")
```

Comment:
This start block will automatically be ended by the nested start block on L65 so if the nested block is ended with a failure then this one will erroneously report success. So I think we should remove this block and incorporate the text in the nested block as suggested below.

Reply:
Seems to be working fine:

```
 ✓ Checking if the metrics are accessible from non-gateway nodes
 ✓ Querying gateway metrics 
 ✓ The gateway metrics are accessible
 ✗ Querying globalnet metrics 
 ✗ Unable to access globalnet metrics service in submariner-operator namespace   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to submariner-globalnet-metrics.submariner-operator.svc.cluster.local port 8081 after 1 ms: Connection refused

 ✗ Failed to access metrics from non-gateway node
```

```
 ✓ Checking if the metrics are accessible from non-gateway nodes
 ✗ Querying gateway metrics 
 ✗ Unable to access gateway metrics service in submariner-operator namespace   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to submariner-gateway-metrics.submariner-operator.svc.cluster.local port 8080 after 1 ms: Connection refused

 ✓ Querying globalnet metrics 
 ✓ The globalnet metrics are accessible
 ✗ Failed to access metrics from non-gateway node
```